content/posts/irods-development-update-february-2023.md

```diff
@@ -0,0 +1,239 @@
+Title: iRODS Development Update: February 2023
+Date: 2023-02-23 12:00
+Author: Kory Draughn
+Slug: irods-development-update-february-2023
+Status: published
+
+
+Hello Everyone,
+
+We're inching closer and closer to a release. Issues are being resolved and polish is being applied. TRiRODS happened. If you're interested in the status of GenQuery2, then you can find the presentation at [https://irods.org/trirods](https://irods.org/trirods).
+
+To start things off, we've finally finished the first draft of the iRODS 2023 Roadmap. If you're a Consortium member, we look forward to your feedback.
+
+The iRODS Consortium will be in Barcelona for CS3. If you're attending the conference, come say hi.
+
+As for software development updates, the server no longer allows empty strings to be stored as metadata attribute values. The client-side and server-side interfaces of the filesystem library and dstream library can now be used within the same translation unit. A new server-side unit testing framework has been developed to improve server-side coverage. The iCommands have also received some improvement. `itouch` now reports a non-zero return code when it fails and `itree` now provides better support for pattern matching.
+
+The Testing Environment now reports the SHA and version of the iRODS server which was just tested. A very helpful quality-of-life improvement. We've restored the ability to run tests on a remote Docker daemon as well.
+
+The Development Environment is receiving some improvements. Work is being done to migrate to new Dockerfile syntax which will enable improved build times of images. We're also investigating ubi images, adding support for qtcreator, and various other things.
+
+A new release of Metalnx is very close. Several issues have been resolved. With the release of iRODS 4.3.0, Metalnx was not able to manipulate metadata. That is now fixed. Metalnx will now create intermediate parent collections automatically. And finally, the search interface correctly handles queries involving fields which span multiple columns.
+
+Users of the Python iRODS Client will be happy to know that it will soon provide similar features and functionality found in `igroupadmin`.
+
+We'll end this month's update on the C++ S3 Bridge. Support for the HeadObject, CopyObject, and DeleteObject operations have been implemented. We're now focused on packaging the software so that users can give it a try. Please stay tuned if you find that interesting.
```

Comment:
we haven't called it 'bridge' in public... I still think it's the C++ S3 API... no?

Reply:
API is more appropriate.

Updated.